cbmc/proofs/mqtt_state.c

```diff
@@ -465,6 +500,48 @@ bool valid_IotMqttSubscriptionList( const IotListDouble_t *pSub,
     length < SUBSCRIPTION_COUNT_MAX;
 }
 
+void *invalid_pointer()
+{
+  void *ptr;
+  return ptr;
+}
+
+void free_IotMqttSubscriptionList( IotListDouble_t *pSub )
+{
+  IotListDouble_t *pThis;
+  IotListDouble_t *pNext;
+
+  if ( pSub == NULL ) return;
+  pThis = pSub->pNext;
+  pSub->pNext = invalid_pointer();
+  pSub->pPrevious = invalid_pointer();
+
+#if 3 < SUBSCRIPTION_COUNT_MAX
```

Comment:
The code in all the three ```#if``` is same. What is the purpose?

Reply:
Same response as in `allocate_IotMqttOperationList`.